### ISSUE TITLE
Different depth wise convolution for different images in the batch

### ISSUE THREAD

**Issue by asellam** [tensorflow/tensorflow#40829]
Hello, I have a situation where I want to apply different depth wise convolutions to different image in a samples batch.
Lets say I have a batch of N images of shape (H x W x C) and N depthwise conv2d filter of shape (K x K x C x 1) I want to apply each one of these N convolutions to each one of the images. Normally, we can do this with tf.unstack along axis 0 on the batch of image and the convolution kernels volume and then process each of images apart using tf.nn.deptwise_conv2d and use tf.concat to build the final result. However, I can't do this cause N is the batch size inside a kerase layer so its value is unkown before compiling the model. Can you suggest a solution or make a version of tf.nn.deptwise_conv2d that supports seperate convolution kernels for separate images?

**Comment by asellam** [comment]
Hello again,

It turns out we can achieve this uisng current Tensorflow operation, I was able to hack this by using tf.unstack on the channels dimension and then using data_format='NCHW':

kernels = tf.expand_dims(kernels , 4) # (C x K x K x N x 1)
kernels = tf.unstack(onehot_2d, axis=0) # [(K x K x N x 1)] * C
samples = tf.unstack(tf.expand_dims(input_tensor, 0), axis=4) # [(1 x N x Hx W)] * C
filtered = [tf.nn.depthwise_conv2d(sample, kernel, [1, 1, 1, 1], 'VALID', data_format='NCHW') 
                 for (sample, kernel) in zip(samples, kernels)] # [(1 x N x P x P)] * C
output = tf.concat(filtered, axis=0) # (C x N x P x P)
return tf.transpose(output, perm=[1, 2, 3, 0]) # (N x P x P x C)
